federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/query/internal/LDAPQuery.java

```diff
@@ -39,16 +43,19 @@
 /**
  * Default IdentityQuery implementation.
  *
+ * LDAPQuery should be closed after use in case that pagination was used (initPagination was called)
  *
  * @author Shane Bryzak
  */
 public class LDAPQuery {
```

Comment:
LDAPQuery should implement AutoCloseable and used via try-with-resources

Reply:
@hmlnarik Thanks for the suggestion and agree it is slightly better. I've updated PR to use AutoCloseable.